src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -3,3 +3,25 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+from ._util import ParametersContext, patch_arg_make_required, patch_arg_update_description
+
+
+with ParametersContext(command='sql server create') as c:
+    from azure.mgmt.sql.models.server import Server
+
+    # About the patches:
+    #
+    # - Both administrator_login and administrator_login_password are required for server creation.
+    # However these two parameters are given default value in the create_or_update function
+    # signature, therefore, they can't be automatically converted to requirement arguments.
+    #
+    # - Due to issue https://github.com/Azure/azure-cli/issues/1644, the description of version
+    # can't be correctly extracted
+    #
+    c.expand('parameters', Server, patches={
+        'administrator_login': patch_arg_make_required,
```

Comment:
> Here's the catch:
> The login and login password are not parameters on the method. Instead, the method accepts a parameter argument which is of type Server. The Server model has the property of these two arguments.
> The arguments are required for create operation, and it is stated in docstring but not elsewhere.

I may not be understanding the process. This method expands the complex object into its constituent parts and registers them as parameters on the command (and handles their reassembly later on)?  So if the constructor for the complex object has default values for these parameters, can you not make those the defaults when you create the parameters by reflecting on the complex object constructor?